osu.Game/Overlays/BeatmapListingOverlay.cs

```diff
@@ -37,6 +41,14 @@ public class BeatmapListingOverlay : FullscreenOverlay<BeatmapListingHeader>
 
         private OverlayScrollContainer resultScrollContainer;
 
+        private readonly IBindable<APIState> apiState = new Bindable<APIState>();
+
+        [Resolved]
+        private IAPIProvider api { get; set; }
+
+        private Container placeholderContainer;
+        private readonly Placeholder errorPlaceholder = new LoginPlaceholder(@"Please sign in to view beatmap listing!");
```

Comment:
This assignment should ideally be inlined in `load()` for readability in all cases.